engine/storage/src/main/java/org/apache/cloudstack/storage/allocator/AbstractStoragePoolAllocator.java

```diff
@@ -179,9 +179,32 @@ public boolean configure(String name, Map<String, Object> params) throws Configu
         } else if(allocationAlgorithm.equals("firstfitleastconsumed")){
             pools = reorderPoolsByCapacity(plan, pools);
         }
+
+        // Hardware accelerated pools are preferred for thick disks
+        if (dskCh != null && !dskCh.getProvisioningType().equals(Storage.ProvisioningType.THIN) &&
+                !storageMgr.DiskProvisioningStrictness.valueIn(plan.getDataCenterId())) {
+            pools = reorderPoolsByDiskProvisioningType(pools, dskCh);
+        }
+
         return pools;
     }
 
+    private List<StoragePool> reorderPoolsByDiskProvisioningType(List<StoragePool> pools, DiskProfile dskCh) {
+        List<StoragePool> reorderedPools = new ArrayList<>();
+        for (StoragePool pool: pools) {
```

Comment:
re-ordering is strictly for thick disk provision type here, i think it should be based on the provisioning type in the disk profile?

Reply:
Yes, If thin disk provisioning is required, reordering is not required, all storage pools support thin disk provisioning.